### DISCUSSION TITLE
How can I set uvicorn to python 3.7?

### DISCUSSION THREAD

**Discussion by gkeniti** [Kludex/uvicorn#1420] [Q&A] [answered]
Hi,

My code requires python >3.7 and it's the default for python3.

But after installing uvicorn it looks using python 3.6.8

 $uvicorn --version
Running uvicorn 0.17.0 with CPython 3.6.8 on Linux

How can I use 3.7 instead?

$ python3 --version
Python 3.7.11
$ which python3
        /usr/local/bin/python3.7

**Comment by euri10** [comment, marked as answer]
you probably want to read more about virtual environments https://docs.python.org/3/tutorial/venv.html
stackoverflow is more suiited for questions like that
